### PR TITLE
refactor: refactor error types in api

### PR DESCRIPTION
**Pull request by jayzhang**

### CODE REVIEW

**Review by Alive-Fish — DISMISSED** [review]

File: packages/fx-core/src/common/m365/launchHelper.ts [hunk 1]
```diff
@@ -95,9 +88,6 @@ export class LaunchHelper {
       }
       return ok(m365AppId);
     } catch (error) {
-      if ((error as FxError).innerError?.response.status === 404) {
```
Comment (swatDong) [inline packages/fx-core/src/common/m365/launchHelper.ts:98]: Add this back. Seems no need to remove this part?

File: packages/fx-core/resource/package.nls.json [hunk 2]
```diff
@@ -632,6 +632,7 @@
   "error.common.MissingRequiredFileError": "Missing %srequired file `%s`",
   "error.common.HttpClientError": "A http client error happened while performing the %s task. The error response is: %s",
   "error.common.HttpServerError": "A http server error happened while performing the %s task. Please try again later. The error response is: %s",
+  "error.common.ConcurrentError": "Previous task is still running. Please wait util your previous task to finish and try again.",
```
Comment (HuihuiWu-Microsoft) [inline packages/fx-core/resource/package.nls.json:635]: Typo 'util'->'until'

File: packages/fx-core/src/common/m365/packageService.ts [hunk 3]
```diff
@@ -100,12 +102,12 @@ export class PackageService {
       } while (true);
     } catch (error: any) {
       this.logger?.error("Sideloading failed.");
-      if (error.response) {
-        this.logger?.error(JSON.stringify(error.response.data));
-        this.traceError(error);
-      } else {
-        this.logger?.error(error.message);
-      }
+      // if (error.response) {
```
Comment (swatDong) [inline packages/fx-core/src/common/m365/packageService.ts:105]: What's the purpose to change the error handling in this file?
Conditional log/trace error is required for m365 action in different runtime, vsc v.s. cli.

**Review by nliu-ms — APPROVED** [review]

File: packages/fx-core/src/common/m365/packageService.ts (hunk 3 above)
Comment (kuojianlu) [inline packages/fx-core/src/common/m365/packageService.ts:105]: Why comment this?

File: packages/fx-core/src/common/m365/packageService.ts [hunk 4]
```diff
@@ -223,12 +223,12 @@ export class PackageService {
       return launchInfo.data;
     } catch (error: any) {
       this.logger?.error("Get LaunchInfo failed.");
-      if (error.response) {
```
Comment (swatDong) [inline packages/fx-core/src/common/m365/packageService.ts:226]: Miss one revert here~

**Review by swatDong — APPROVED** [review]

**Review by nliu-ms — APPROVED** [review]

**Review by nliu-ms — APPROVED** [review]

**Review by Alive-Fish — APPROVED** [review]

**Review by HuihuiWu-Microsoft — APPROVED** [review]

**Review by yukun-dong — APPROVED** [review]

**Review by xzf0587 — APPROVED** [review]

**Review by Siglud — APPROVED** [review]

File: packages/fx-core/resource/package.nls.json [hunk 5]
```diff
@@ -632,6 +632,7 @@
   "error.common.MissingRequiredFileError": "Missing %srequired file `%s`",
   "error.common.HttpClientError": "A http client error happened while performing the %s task. The error response is: %s",
   "error.common.HttpServerError": "A http server error happened while performing the %s task. Please try again later. The error response is: %s",
+  "error.common.ConcurrentError": "Previous task is still running. Please wait until your previous task to finish and try again.",
```
Comment (sffamily) [inline packages/fx-core/resource/package.nls.json:635]: Please -> remove
to finish -> is finished

Comment (jayzhang) [reply packages/fx-core/resource/package.nls.json:635]: Will fix in another PR.

**Review by sffamily — APPROVED** [review]

**Pull request merged**